mypyc/irbuild/mapper.py

```diff
@@ -132,7 +132,11 @@ def fdef_to_sig(self, fdef: FuncDef) -> FuncSignature:
         else:
             # Handle unannotated functions
             arg_types = [object_rprimitive for arg in fdef.arguments]
-            ret = object_rprimitive
+            # We at least know the return type for __init__ will be None.
+            if fdef.name == '__init__':
```

Comment:
Could `fdef` be a module-level function with the (confusing) name `__init__`? Add test for this. We should allow top-level functions to return any values, even if they are named `__init__` for some reason. 

(You should be able to compare the `FuncDef` `info` attribute against `FUNC_NO_INFO` to check whether it's a method.)

Reply:
Oh nice catch this will not work for module-level functions named `__init__`! I will add a check and test for this. Thank you for the `FUNC_NO_INFO` tip.